src/Data/Nat/GCD.agda

```diff
@@ -103,11 +106,41 @@ gcd[m,n]≡0⇒n≡0 : ∀ m {n} → gcd m n ≡ 0 → n ≡ 0
 gcd[m,n]≡0⇒n≡0 m {zero}  eq = P.refl
 gcd[m,n]≡0⇒n≡0 m {suc n} eq = contradiction eq (gcd[m,n]≢0 m (suc n) (inj₂ λ()))
 
-gcd-comm : ∀ m n → gcd m n ≡ gcd n m
+gcd-comm : Commutative gcd
 gcd-comm m n = ∣-antisym
   (gcd-greatest (gcd[m,n]∣n m n) (gcd[m,n]∣m m n))
   (gcd-greatest (gcd[m,n]∣n n m) (gcd[m,n]∣m n m))
 
+gcd-assoc : Associative gcd
+gcd-assoc m n p = ∣-antisym
+  (gcd-greatest {m} {gcd n p} (∣-trans (gcd[m,n]∣m (gcd m n) p) (gcd[m,n]∣m m n))
```

Comment:
Wouldn't this proof be much more readable if written equationally?

Reply:
Have a look at `∣-Reasoning`

Reply:
I may be missing something. How would equational reasoning help here? As I understand it, it's another way to write `|-trans`, but it is not being chained here. IMO the "creative" part of the proof is the application of `|-antisym` and `gcd-greatest` at the beginning, so it doesn't seem worth drowning it out by expanding the bits at the leaves of the proof, or is there really a more equational way to phrase this proof?

Reply:
`|-trans` **is** chaining.

Reply:
I thought there were other proofs further down that used `|-trans` for more than 2 steps. Also, I often find that it's worth extracting an inner proof like this into a lemma.

I don't insist that this particular proof use equational reasoning, just that you carefully look at all your proofs and see if some of them could be easier to understand (by future readers!) if they were done equationally.

Reply:
Good idea. If I put those in lemmas then (I hope) everyone will be happy. (done)